examples/exampleSetup.hpp

```diff
@@ -118,49 +118,72 @@ namespace example {
-
-        using FDT = Opm::FlowDiagnostics::Toolbox;
-
-        auto fl = extractFluxField(G, step);
-        const size_t num_conn = fl.numConnections();
-        const size_t num_phases = fl.numPhases();
-        for (size_t conn = 0; conn < num_conn; ++conn) {
-            using Co = Opm::FlowDiagnostics::ConnectionValues::ConnID;
-            using Ph = Opm::FlowDiagnostics::ConnectionValues::PhaseID;
-            for (size_t phase = 0; phase < num_phases; ++phase) {
-                fl(Co{conn}, Ph{phase}) /= 86400; // HACK! converting to SI.
-            }
-        }
-
         Opm::FlowDiagnostics::CellSetValues inflow;
         for (const auto& well : well_fluxes) {
             for (const auto& completion : well.completions) {
                 const int grid_index = completion.grid_index;
                 const auto& ijk = completion.ijk;
                 const int cell_index = G.activeCell(ijk, grid_index);
-                inflow.addCellValue(cell_index, completion.reservoir_inflow_rate);
+                if (cell_index >= 0) {
+                    inflow.addCellValue(cell_index, completion.reservoir_inflow_rate);
+                }
+            }
+        }
+
+        return inflow;
+    }
+
+    namespace Hack {
+        inline Opm::FlowDiagnostics::ConnectionValues
+        convert_flux_to_SI(Opm::FlowDiagnostics::ConnectionValues&& fl)
+        {
+            using Co = Opm::FlowDiagnostics::ConnectionValues::ConnID;
+            using Ph = Opm::FlowDiagnostics::ConnectionValues::PhaseID;
+
+            const auto nconn = fl.numConnections();
+            const auto nphas = fl.numPhases();
+
+            for (auto phas = Ph{0}; phas.id < nphas; ++phas.id) {
+                for (auto conn = Co{0}; conn.id < nconn; ++conn.id) {
+                    fl(conn, phas) /= 86400;
+                }
             }
+
+            return fl;
         }
+    }
+
+    inline Opm::FlowDiagnostics::Toolbox
+    initialiseFlowDiagnostics(const Opm::ECLGraph& G)
+    {
+        const auto connGraph = Opm::FlowDiagnostics::
+            ConnectivityGraph{ static_cast<int>(G.numCells()),
+                               G.neighbours() };
 
         // Create the Toolbox.
-        auto tool = FDT{ connGraph };
+        auto tool = Opm::FlowDiagnostics::Toolbox{ connGraph };
+
         tool.assignPoreVolume(G.poreVolume());
-        tool.assignConnectionFlux(fl);
-        tool.assignInflowFlux(inflow);
+        tool.assignConnectionFlux(Hack::convert_flux_to_SI(extractFluxField(G)));
+
+        auto wsol = Opm::ECLWellSolution{};
+
+        const auto well_fluxes =
+            wsol.solution(G.rawResultData(), G.numGrids());
+
+        tool.assignInflowFlux(extractWellFlows(G, well_fluxes));
 
         return tool;
     }
 
 
-    Opm::FlowDiagnostics::Toolbox setup(int argc, char* argv[])
+    inline auto setup(int argc, char* argv[])
+        -> decltype(initialiseFlowDiagnostics(std::declval<Opm::ECLGraph>()))
```

Comment:
I believe this sort of declaration, while correct, is rather unreadable and should be avoided in non-generic code. (Referring to the use of decltype and declval, not trailing return type.)